### PR TITLE
adds support for max_replicas_per_node and placement preferences

### PR DESCRIPTION
**Pull request by dvallant**

### CODE REVIEW

File: client/src/main/groovy/de/gesellix/docker/client/stack/DeployConfigReader.groovy [hunk 1]
```diff
@@ -761,4 +764,14 @@ class DeployConfigReader {
...
         return configSpec
     }
+
+    List<PlacementPreferences> placementPreferences(List<de.gesellix.docker.compose.types.PlacementPreferences> preferences) {
+        log.info("placementPreferences: ${preferences}")
+        if (preferences == null) {
+            return null
+        }
+        def spread = new Spread(spreadDescriptor: preferences[0].spread)
+        log.info( "spread: ${spread}" )
```
Comment (gesellix) [inline client/src/main/groovy/de/gesellix/docker/client/stack/DeployConfigReader.groovy:774]: Can you adjust the code formatting to be consistent with the surrounding code, please?
```suggestion
        log.info("spread: ${spread}")
```

File: client/src/main/groovy/de/gesellix/docker/client/stack/DeployConfigReader.groovy [hunk 2]
```diff
@@ -761,4 +764,14 @@ class DeployConfigReader {
...
+        return [new PlacementPreferences( spread: spread )]
```
Comment (gesellix) [inline client/src/main/groovy/de/gesellix/docker/client/stack/DeployConfigReader.groovy:775]:
```suggestion
        return [new PlacementPreferences(spread: spread)]
```

File: client/src/main/groovy/de/gesellix/docker/client/stack/DeployConfigReader.groovy (hunk 1 above)
Comment (dvallant) [reply client/src/main/groovy/de/gesellix/docker/client/stack/DeployConfigReader.groovy]: oh yeah, haven't seen it before...

**Comment by gesellix** [comment]
The new release is available as version `2020-06-19T16-12-55`. Thank you!

**Comment by dvallant** [comment]
> The new release is available as version `2020-06-19T16-12-55`. Thank you!

Thank you too!

**Pull request merged**